Fix this.connected bug when connect event is missed

willyjchen opened the pull request:
This fix a bug that this.connected is not initialized if the redis connect event is missed.

Check if the connect event is missed or not, then emit the missed connect event again.

willyjchen committed:
Fix this.connected bug when connect event is missed

This fix a bug that this.connected is not initialized if the redis connect event is missed.

Check if the connect event is missed or not, then emit the missed connect event again.

atn18 commented on lib/ExpressRedisCache.js:
```diff
@@ -96,6 +96,10 @@ module.exports = (function () {
       }.bind(this));
+      
+      if ( this.client.connected ) {
+        this.client.emit('connect');
```
I think resending `connect` on `client` is not good idea. Because there could be some subscribers that will not expect this.
You should better directly set `self.connected`.

willyjchen committed:
Update ExpressRedisCache.js

1. Avoid resenting redis 'connect' event as suggested change
2. Add check of 'this.connected' to prevent init twice
